lib/bin/lwm2m_carrier/lwm2m_carrier.rst

```diff
@@ -3,24 +3,160 @@
 #############
 
-You can use the **LwM2M carrier** library in your nRF91 application in order to connect to the operator LwM2M network.
-Integrating this library into an application causes the device to connect to the device management platform autonomously.
+You can use the LwM2M carrier library in your nRF91 application in order to connect to the LwM2M operator network.
+Integrating this library into an application causes the device to connect to the operator's device management platform autonomously.
 
-Every released version is considered for certification with carriers.
-Refer to the :ref:`liblwm2m_carrier_changelog` to check the certification status of a particular release.
-Your final device must go through certification by the carrier.
+Some examples of carrier device management platforms are listed below:
 
-The :ref:`lwm2m_carrier` sample demonstrates how to run this library in an application.
+* `Verizon's Thingspace`_
+* `AT&T's IoT Platform`_
+
+This library uses LwM2M protocol to communicate with device manager platforms, but it does not expose an LwM2M API to your application. If you want to use LwM2M for other purposes, see :ref:`lwm2m_interface`.
 
-Prerequisites
-*************
-The LwM2M carrier library has dependencies on Zephyr and the |NCS|.
-See the :ref:`liblwm2m_carrier_changelog` of the library to check which version of |NCS| it is built for.
-To integrate the library into an application, the application must be based on a supported |NCS| version.
+The :ref:`lwm2m_carrier` sample demonstrates how to run this library in an application.
+The LwM2M carrier library is also used in the :ref:`asset_tracker` application.
 
 Application integration
 ***********************
+
+The LwM2M carrier library has an OS abstraction layer.
+See :file:`lwm2m_os.h`.
+This makes the LwM2M carrier library independent of the underlying implementation of primitives such as timers, non-volatile storage, and heap allocation.
+The OS abstraction is fully implemented for the |NCS|, and it would have to be ported if used on other systems.
+
 To run the library in an application, you must implement the application with the API of the library.
+You can enable the module using the :option:`CONFIG_LWM2M_CARRIER` Kconfig option.
+
+The :ref:`lwm2m_carrier` sample project configuration (:file:`nrf/samples/nrf9160/lwm2m_carrier/prj.conf`) contains all the configurations that are needed by the LwM2M carrier library.
+
+A bootstrap server provides the application with the configuration needed to connect to the device management servers.
+During self-testing of the certification process, you can provide the initialization parameter :cpp:type:`lwm2m_carrier_config_t` to specify the `Pre-Shared Key (PSK)`_, and the URI of the bootstrap server.
+This can be done by using the :option:`CONFIG_LWM2M_CARRIER_USE_CUSTOM_BOOTSTRAP_URI`, :option:`CONFIG_LWM2M_CARRIER_CUSTOM_BOOTSTRAP_URI`, and :option:`CONFIG_LWM2M_CARRIER_USE_CUSTOM_BOOTSTRAP_PSK` Kconfig options.
```

Comment:
Let's mention the `certification_mode` option here, too.

Reply:
Actually, that is added (correcty) by the PR with release `0.9.0`, since it applies from that version.